lib/components/SelectionControl/radio/radio-button/index.tsx

```diff
@@ -0,0 +1,116 @@
+import React from "react";
+import clsx from "clsx";
+import {
+    LabelPairedCircleMdRegularIcon,
+    LabelPairedCircleDotMdFillIcon,
+    LabelPairedCircleInfoMdRegularIcon,
+} from "@deriv/quill-icons";
+import { Text } from "../../../Typography";
+import "./radio-button.scss";
+
+type TRadio = {
+    className?: string;
+    classNameInfo?: string;
+    classNameLabel?: string;
+    defaultChecked?: boolean;
+    disabled?: boolean;
+    has_info?: boolean;
+    id?: string;
+    name?: string;
+    onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+    required?: boolean;
+    value?: string;
+};
+
+const RadioButton = ({
+    children,
+    className,
+    classNameInfo,
+    classNameLabel,
+    defaultChecked,
+    disabled = false,
+    has_info,
+    id,
+    onChange,
+    value,
+    ...otherProps
+}: React.PropsWithChildren<TRadio>) => {
+    const [checked, setChecked] = React.useState(defaultChecked);
+
+    React.useEffect(() => {
+        setChecked(defaultChecked);
+    }, [defaultChecked]);
+
+    const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setChecked(e.target.checked);
+        onChange && onChange(e);
+    };
+
+    const handleIconClick = (e: React.MouseEvent<HTMLSpanElement>) => {
+        if (!disabled) {
+            const synthesizedEvent = new Event("change", { bubbles: true });
```

Comment:
Am I'm right that when user clicks on `span`, `input` should catch `onChange` event? If yes, maybe we can use `<label>` instead of `span`? In this case you can bind them with `name` or `id`, so whenever user clicks on `label`, `input` will be checked. Also, there is another way: you can position `span` just over the `input` and make span non-interactive (`pointer-events: none;`), so actually, user will react with input inself

Reply:
This is not specifically for the span or label. There's a custom icon for input, whenever it's clicked I trigger the hidden input to trigger logic changes across all radio buttons across the group. The sync between all the buttons is done via the hidden input.

Reply:
I mean you can still render `input`, but removing `display: none` and hide by positioning icon over it. If you add `pointer-events: none;` to the icon, user will interact with `input` itself, so in that case there will be no need in extra logic

Reply:
Hmm, this sounds too fragile solution to me. If something goes wrong with the layout, the input will be duplicated on the screen which is not a good UX. I might be misunderstanding what you mean though, but I think hiding input is a better option here

Reply:
Okay👍